spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/MqttPahoMessageDrivenChannelAdapter.java

```diff
@@ -379,4 +399,46 @@ public void messageArrived(String topic, MqttMessage mqttMessage) {
 	public void deliveryComplete(IMqttDeliveryToken token) {
 	}
 
+	/**
+	 * Used to complete message arrival when {@link AckMode#MANUAL}.
+	 *
+	 * @since 5.3
+	 */
+	private static class AcknowledgmentImpl implements SimpleAcknowledgment {
```

Comment:
I wonder if this internal one should fully implement a `AcknowledgmentCallback` fro compatibility with a `AcknowledgmentCallback getAcknowledgmentCallback(Message<?> message)`  contract...

Any objection do not do that?

Thanks

Reply:
That is intended for message source implementations. We can't support its other features here.

Reply:
OK. We an give it a shot.
But let's document it in that docs properly, that this MQTT ack is not an `AcknowledgmentCallback`.

Also I wonder if we should say something on the matter in the `polling-consumer.adoc#deferred-acks-message-source` ...

Reply:
I will add some more docs...